metabase/metabase
# Convert ParameterSidebar to TypeScript

Epic https://github.com/metabase/metabase/issues/26898

How to test:
- Create a dashboard with several questions, parameters and map them to questions
- Edit a dashboard and click on the parameter. The parameter sidebar should be opened.
- Make sure it is possible to edit all the fields in the sidebar.
- Make sure it is possible to remove the parameter in the sidebar.
- Go to `Linked filters` tab in the sidebar.
- Make sure that the toggle is working.
- Make sure it is possible to click on the parameter name and see its connected fields.

Notes:
- I'm very happy about this because I now know there is a huge difference between `Parameter` and `UiParameter` and one cannot be used easily in place of another. TypeScript for the rescue. 

<img width="1004" alt="Screenshot 2022-12-14 at 19 01 21" src="https://user-images.githubusercontent.com/8542534/207659718-a0d2102f-9559-4bca-ba0d-2128ad38af7b.png">
<img width="999" alt="Screenshot 2022-12-14 at 19 08 19" src="https://user-images.githubusercontent.com/8542534/207661153-9c0294b9-dcd5-4565-9dcf-e0453228b368.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27170)
